python/lsst/meas/astrom/matcher_probabilistic.py

```diff
@@ -601,24 +632,28 @@ def match(
                 n_finite = np.sum(finite, axis=1)
                 # Require some number of finite chi_sq to match
                 chisq_good = n_finite >= config.match_n_finite_min
-                if np.any(chisq_good):
-                    try:
-                        chisq_sum = np.zeros(n_found, dtype=float)
-                        chisq_sum[chisq_good] = np.nansum(chi[chisq_good, :] ** 2, axis=1)
-                        idx_chisq_min = np.nanargmin(chisq_sum / n_finite)
-                        ref_match_meas_finite[index_row] = n_finite[idx_chisq_min]
-                        ref_match_count[index_row] = len(chisq_good)
-                        ref_chisq[index_row] = chisq_sum[idx_chisq_min]
-                        idx_match_select = found[idx_chisq_min]
-                        row_target = target.extras.indices[idx_match_select]
-                        ref_row_match[index_row] = row_target
-
-                        target_row_match[row_target] = index_row
-                        matched_target.add(idx_match_select)
-                    except Exception as error:
-                        # Can't foresee any exceptions, but they shouldn't prevent
-                        # matching subsequent sources
-                        exceptions[index_row] = error
+                if not any(chisq_good):
+                    continue
+                try:
+                    chisq_sum = np.zeros(n_found, dtype=float)
+                    chisq_sum[chisq_good] = np.nansum(chi[chisq_good, :] ** 2, axis=1)
+                    idx_chisq_min = np.nanargmin(chisq_sum / n_finite)
```

Comment:
Are we always safeguarded against division by zero here?

Reply:
As long as n_finite_min > 0, yes.

Reply:
Yes, that's why I suggested you add `check` in `match_n_finite_min = pexConfig.Field(...)`.